mobile/lib/features/wallet/balance_row.dart

```diff
@@ -48,28 +47,26 @@ class _BalanceRowState extends State<BalanceRow> with SingleTickerProviderStateM
 
     PositionChangeNotifier positionChangeNotifier = context.watch<PositionChangeNotifier>();
 
-    Amount amount;
-    String name;
-    Color rowBgColor;
-    SvgPicture icon;
-
-    if (widget.walletType == WalletType.lightning) {
-      name = "Lightning";
-      rowBgColor = theme.lightning;
-      icon = SvgPicture.asset("assets/Lightning_logo.svg");
-      amount = walletChangeNotifier.lightning();
-    } else if (widget.walletType == WalletType.stable) {
-      name = "Synthetic-USD";
-      rowBgColor = theme.lightning;
-      icon = SvgPicture.asset("assets/USD_logo.svg");
-      // this in unused for now, other than to initialise value (and satisfy the compiler)
-      amount = positionChangeNotifier.getStableUSDAmountInSats();
-    } else {
-      name = "On-chain";
-      rowBgColor = theme.onChain;
-      icon = SvgPicture.asset("assets/Bitcoin_logo.svg");
-      amount = walletChangeNotifier.onChain();
-    }
+    final (name, rowBgColor, icon, amount) = switch (widget.walletType) {
+      WalletType.lightning => (
+          "Lightning",
+          theme.lightning,
+          SvgPicture.asset("assets/Lightning_logo.svg"),
+          walletChangeNotifier.lightning(),
+        ),
+      WalletType.onChain => (
+          "On-chain",
+          theme.onChain,
+          SvgPicture.asset("assets/Bitcoin_logo.svg"),
+          walletChangeNotifier.onChain(),
+        ),
+      WalletType.stable => (
+          "Synthetic-USD",
+          theme.lightning,
+          SvgPicture.asset("assets/USD_logo.svg"),
+          positionChangeNotifier.getStableUSDAmountInSats()
+        ),
+    };
```

Comment:
cool, I didn't know this syntax is possible in dart.

Reply:
Yea, it's really nice :) - called 'switch expressions'